app/assets/javascripts/actions/projectBoard.js

```diff
@@ -53,3 +79,31 @@ export const fetchProjectBoard = projectId =>
       return dispatch(errorRequestProjectBoard(error));
     }
   };
+
+export const closeSearch = () =>
+  dispatch => {
+    dispatch(closeSearchSuccess());
+    dispatch(receiveStories([], storyScopes.SEARCH));
+  }
+
+export const search = (keyWord, projectId) =>
+  async (dispatch, _, { Story, ProjectBoard }) => {
+    if (ProjectBoard.validSearch(keyWord)) {
```

Comment:
Same here, we should move it to the Search model but we should not do this conditional this way. Remember when we talked about tell-dont-ask? What do you think about this:

```js
Search.searchStories(keyWorld, {
  onStart: () => dispatch(updateLoadingSearch(true)),
  onSuccess: (results) => {
    dispatch(updateLoadingSearch(false));
    dispatch(searchStoriesSuccess(keyWord));
    dispatch(receiveStories(result, 'search'));
    sendErrorNotificationIfNeeded(dispatch, 'projects.stories_not_found', result.length);
  },
  onError: (error) => {
    dispatch(sendErrorNotification('messages.operations.error.default_error', { custom: true }));
    dispatch(updateLoadingSearch(false));
    console.error(error)
  }
});
```